managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedLedgerImpl.java

```diff
@@ -2182,63 +2180,33 @@ public boolean hasMoreEntries(PositionImpl position) {
         return result;
     }
 
-    void discardEntriesFromCache(ManagedCursorImpl cursor, PositionImpl newPosition) {
-        Pair<PositionImpl, PositionImpl> pair = activeCursors.cursorUpdated(cursor, newPosition);
-        if (pair != null) {
-            entryCache.invalidateEntries(pair.getRight());
+    void doCacheEviction(long maxTimestamp) {
+        if (entryCache.getSize() > 0) {
+            entryCache.invalidateEntriesBeforeTimestamp(maxTimestamp);
```

Comment:
It is no more necessary to call it in `doCacheEviction` method since `invalidateEntriesUpToSlowestReaderPosition` will get called when the slowest reader position changes.

Reply:
Oh, got it. Thanks.